code/modules/mob/living/basic/ruin_defender/mimic/mimic.dm

```diff
@@ -0,0 +1,381 @@
+	return ..()
+
+/mob/living/basic/mimic/crate/attack_hand(mob/living/carbon/human/user, list/modifiers)
+	if(user.combat_mode)
+		return ..()
+	if(trigger())
+		to_chat(user, span_danger("As you try to open [src] it stiffens up and nearly clamps down on your fingers!"))
+		return TRUE
+	toggle_open(user)
+	return TRUE
+
+/mob/living/basic/mimic/crate/melee_attack(mob/living/carbon/target, list/modifiers, ignore_cooldown)
+	. = ..()
+	toggle_open() // show our cool lid at the dumbass humans
+
+/mob/living/basic/mimic/crate/proc/trigger()
+	if(isnull(ai_controller) || client)
+		return FALSE
+	if(ai_controller.ai_status != AI_STATUS_OFF)
+		return FALSE
+	visible_message(span_danger("[src] starts to move!"))
+	REMOVE_TRAIT(src, TRAIT_AI_PAUSED, INNATE_TRAIT)
+	ai_controller.set_ai_status(AI_STATUS_ON)
+	if(length(contents))
+		locked = TRUE //if this was a crate with loot then we dont want people to just leftclick it to open it then bait it somewhere and steal its loot
+	return TRUE
+
+/mob/living/basic/mimic/crate/adjust_health(amount, updating_health = TRUE, forced = FALSE)
+	if(amount > 0)
+		trigger()
+	return ..()
+
+/mob/living/basic/mimic/crate/death()
+	var/obj/structure/closet/crate/C = new(get_turf(src))
+	// Put loot in crate
+	for(var/obj/O in src)
+		O.forceMove(C)
+	return ..()
+
+/mob/living/basic/mimic/crate/early_melee_attack(atom/target, list/modifiers, ignore_cooldown)
+	if(target == src)
+		toggle_open()
+		return FALSE
+	return ..()
+
+/mob/living/basic/mimic/crate/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(istype(mover, /obj/structure/closet))
+		return FALSE
+/**
+* Used to open and close the mimic
+*
+* Will insert tile contents into the mimic when closing
+* Will dump mimic contents into the time when opening
+* Does nothing if the mimic locked itself
+*/
+/mob/living/basic/mimic/crate/proc/toggle_open(mob/user)
+	if(locked)
+		if(user)
+			balloon_alert(user, "too stiff!")
+		return
+	if(!opened)
+		ADD_TRAIT(src, TRAIT_UNDENSE, MIMIC_TRAIT)
+		opened = TRUE
+		icon_state = "crateopen"
+		playsound(src, 'sound/machines/crate/crate_open.ogg', 50, TRUE)
+		for(var/atom/movable/movable as anything in src)
+			movable.forceMove(loc)
+	else
+		REMOVE_TRAIT(src, TRAIT_UNDENSE, MIMIC_TRAIT)
+		opened = FALSE
+		icon_state = "crate"
+		playsound(src, 'sound/machines/crate/crate_close.ogg', 50, TRUE)
+		for(var/atom/movable/movable as anything in get_turf(src))
+			if(movable != src && insert(movable) == -1)
+				playsound(src, 'sound/items/trayhit/trayhit2.ogg', 50, TRUE)
+				break
+/**
+* Called by toggle_open to put items inside the mimic when it's being closed
+*
+* Will return -1 if the insertion fails due to the storage capacity of the mimic having been reached
+* Will return FALSE if insertion fails
+* Will return TRUE if insertion succeeds
+* Arguments:
+* * AM - item to be inserted
+*/
+/mob/living/basic/mimic/crate/proc/insert(atom/movable/movable)
+	if(contents.len >= storage_capacity)
+		return -1
```

Comment:
Make this `-1` a define so you don't need to read the comment to know what it means.